cloudprovider/cloudprovider.go

```diff
@@ -100,7 +100,11 @@ func GetInstance(cloud Interface, IP string) (instance *types.Instance, err erro
 	runningMutex.Lock()
 	running[IP] = time.Now()
 	runningMutex.Unlock()
-	defer func() { delete(running, IP) }()
+	defer func() {
```

Comment:
Nah :D, Golang maps are not safe for concurrent writes, they're fine for read. I was saying that because I just finished [the Little Go Book](http://openmymind.net/assets/go/go.pdf) and it says otherwise ;)

> The only concurrent thing you can safely do to a variable is to read from it. You can have as many readers as you want, but writes need to be synchronized.

That's why there is another type of mutex, the [RWMutex](https://golang.org/pkg/sync/#RWMutex) if you want to synchronize reads but bad things won't happen because of concurrency only.